R/create_single_dose_dataset.R

```diff
@@ -364,9 +363,15 @@ create_single_dose_dataset <- function(dataset,
     rename(!!dose_freq := !!lookup_column)
 
   # Check that NAs do not appear in start_date or start_datetime or end_date or end_datetime columns
+  condition <- paste0("is.na(", as_name(start_date), ") | is.na(", as_name(end_date), ")")
+  if (!quo_is_null(start_datetime)) {
+    condition <- paste(condition, "| is.na(", as_name(start_datetime), ")")
+  }
+  if (!quo_is_null(end_datetime)) {
+    condition <- paste(condition, "| is.na(", as_name(end_datetime), ")")
+  }
   na_check <- dataset %>%
-    filter(is.na(!!start_date) | is.na(!!end_date) |
-      is.na(!!start_datetime) | is.na(!!end_datetime)) %>%
+    filter(!!parse_expr(condition)) %>%
     select(!!start_date, !!end_date, !!start_datetime, !!end_datetime)
```

Comment:
This is more me not being as comfortable with quosures but by making the `_datetime` arguments optional, will these respective columns always exist/this line always run properly?

Reply:
`select()` ignores arguments which are `NULL`. For the default values the arguments resolve to `select(ASTDT, AENDT, NULL, NULL)` which selects `ASTDT` and `AENDT`.